src/com/owncloud/android/MainApp.java

```diff
@@ -74,6 +77,11 @@ public void onCreate(){
         // initialise thumbnails cache on background thread
         new ThumbnailsCacheManager.InitDiskCacheTask().execute();
         
+        Log_OC.d("InstantUploadFolderObserverService", "start Service InstantUploadFolderObserverService");
+        //startService(new Intent(MainApp.getAppContext(), InstantUploadFolderObserverService.class));
+        Intent i = new Intent(this, InstantUploadFolderObserverService.class);
```

Comment:
This is a good place to start the observer service, better than the current call in ```FileDisplayActivity#onCreate``` for ```FileObserverService```. Kudos for you.

Additionally, a good place to add it too is ```BootUpBroadcastReceiver```, to get it's started as soon as the device ends its boot.